src/metrics/metrics.jl

```diff
@@ -462,15 +337,19 @@ e.g., X[species=1:6] is Taxa 1, size classes 1-6; X[species=7:12] is Taxa 2, siz
 # Arguments
 - `X` : raw results (proportional coral cover relative to full site area)
 - `n_species` : number of species (taxa and size classes) considered
-- `scen` : scenario number to calculate metric for
 - `colony_vol_m3_per_m2` : estimated cubic volume per m² of coverage for each species/size class (36)
 - `max_colony_vol_m3_per_m2` : theoretical maximum volume per m² of coverage for each taxa (6)
-- `site_area` : total area of site in m²
 - `k_area` : habitable area of site in m² (i.e., `k` area)
 """
-function _shelter_species_loop(X::AbstractArray{T1,3}, n_species::Int64, colony_vol_m3_per_m2::Array{F}, max_colony_vol_m3_per_m2::Array{F}, k_area::Array{F})::NamedDimsArray where {T1<:Real,F<:Float64}
+function _shelter_species_loop(
+    X::AbstractArray{T1,3},
+    n_species::Int64,
+    colony_vol_m3_per_m2::Array{F},
+    max_colony_vol_m3_per_m2::Array{F},
+    k_area::Array{F}
+)::YAXArray where {T1<:Real,F<:Float64}
     # Calculate absolute shelter volumes first
-    ASV = NamedDimsArray{(:timesteps, :species, :sites)}(zeros(size(X)...))
+    ASV::YAXArray = DataCube(zeros(size(X)...), (:timesteps, :species, :sites))
```

Comment:
Should we add an equivalent ZeroDataCube function with axes lists like DataCube?

Reply:
I've just added a new ZeroDataCube constructor and updated the old one. You might need to update some code (sorry for that).